### DISCUSSION TITLE
Duda sobre valgrind ./exc

### DISCUSSION THREAD

**Discussion by aangel-sama** [IIC2133-PUC/2024-2#164] [Tarea 2] [answered]
Hola, estaba ejecutando el  valgrind ./secretfind en mi cmd, para ver los leaks de memoria y me salio esto:
![image](https://github.com/user-attachments/assets/8b5489e3-b823-47b2-8232-47813b66f44d)
Asumo, que no tengo fugas porque he liberado todo, pero me causa duda los WARNING, que aparecen ahí, no logro interpretarlo. Esto me pasa en la parte 2. Algo en tomar en cuenta es que en el Makefile, modifique algo para que el tiempo de ejecución fuera más rapido.
![image](https://github.com/user-attachments/assets/bc2c0db4-4dc9-4bc0-ad7e-3171d44b4c2b)
No sé si eso tenga algo que ver. Gracias

**Comment by sebacarrascop** [comment, marked as answer]
Al usar la opción -O3 en tu Makefile, aplica optimizaciones muy agresivas. Estas optimizaciones reestructuran el código para mejorar el rendimiento, pero a veces pueden confundir a Valgrind. Tienes el mismo problema con la otra flag?

**Reply by aangel-sama** [reply]
Si
